core/DataAccess/Model.php

```diff
@@ -97,7 +100,7 @@ public function getInvalidatedArchiveIdsSafeToDelete($archiveTable)
             // if there is more than one archive, the older invalidated ones can be deleted
             if (!empty($duplicateArchives)) {
```

Comment:
Does this additional check/loop still makes sense? Assuming we have some values in `$duplicateArchives` the `while` above should remove all values from that variable, unless there is a broken one in the end. But even with the broken one, the loop below would only do some logging an break.

Reply:
@sgiehl yes it's still needed, the first loop removes the latest usable archive entries from the array. This can be DONE_PARTIAL entries as well as the latest DONE_OK entry. The second loop just collects the remaining values for purging. There might be a way to merge them but I suspect it would get confusing.

Reply:
Ah ok, makes sense. I think I got it now 🙈